python/seldon_core/user_model.py

```diff
@@ -0,0 +1,239 @@
+from seldon_core.metrics import validate_metrics
+from seldon_core.flask_utils import SeldonMicroserviceException
+import json
+from typing import Dict, List, Union
+import numpy as np
+
+def client_custom_tags(user_model: object) -> Dict:
+    """
+    Get tags from user model
+    Parameters
+    ----------
+    user_model
+
+    Returns
+    -------
+       Dictionary of key value pairs
+
+    """
+    if hasattr(user_model, "tags"):
+        return user_model.tags()
+    else:
+        return {}
+
+
+def client_class_names(user_model: object, predictions: np.ndarray) -> List[str]:
+    '''
+    Get class names from user model
+    Parameters
+    ----------
+    user_model
+       User defined class instance
+    predictions
+       Prediction results
+    Returns
+    -------
+       Class names
+    '''
+    if len(predictions.shape) > 1:
+        if hasattr(user_model, "class_names"):
+            return user_model.class_names
+        else:
+            n_targets = predictions.shape[1]
+            return ["t:{}".format(i) for i in range(n_targets)]
```

Comment:
"t" for target in the output is not quite obvious unless you read the code, especially when the name of the function is `client_class_names`